src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,53 @@
+package duke.task;
+
+import duke.DukeException;
+
+import java.io.Serializable;
+
+/**
+ * This class provides the skeleton for a Duke Task.
+ * To implement this class, the programmer only needs to extend this class and provide an implementation for
+ * the {@link #toString()} method.
+ */
+public abstract class Task implements Serializable {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String d) {
+        description = d;
+        isDone = false;
```

Comment:
isDone can be assigned immediately upon declaration e.g `protected boolean isDone = false;`